libbeat/kibana/index_pattern_generator.go

```diff
@@ -65,7 +67,7 @@ func (i *IndexPatternGenerator) generate5x(fields common.Fields) (string, error)
 		return "", err
 	}
 
-	file5x := filepath.Join(i.targetDir5x, i.targetFilename)
+	file5x := filepath.Join(i.targetDir, i.targetFilename)
```

Comment:
We should remove line 64:
```
version, _ := common.NewVersion("5.0.0")
```

Rather use the given version, otherwise you don't really use the provided version from outside for fetching versionized information from the fields.

When generalizing this, the `generate5x` and `generate6x` methods have more overlap and might be a bit restructured.

Reply:
I definitively agree on this one and think the methods should probably be merge in the long run. But I suggest to do this in a follow up PR to not have too many changes in here.

Reply:
Agree on doing the method cleanup in a next version. 
Not passing in the version here but hardcoding it, I consider to be a bug though, e.g. one could pass in `7.0.0` which might have differences in the fields.yml to `6.0.0` but would still get the values for `6.0.0` with this handling.

Reply:
I changed it to `i.version`. I agree with 7.0 it would not be the expected behaviour but this is not a "bug" that was introduced here but already in the code.

Reply:
The difference is that so far no version was given from the outside, but it was hardcoded inside the generator. Thus it was clear that it gets generated for those two defined versions.

This PR adds the possibility to pass in a version. The "bug" I was referring to would be to not use this version then in all places.